include/zephyr/drivers/fuel_gauge.h

```diff
@@ -182,4 +227,6 @@ __subsystem struct fuel_gauge_driver_api {
 }
 #endif /* __cplusplus */
 
+#include <syscalls/fuel_gauge.h>
```

Comment:
This is where 'struct fuel_gauge_get_property' is defined, right? It's referenced already above.

Reply:
`<syscalls/fuel_gauge.h>` references code that gets generated later that is not part of this head.

Example:

```
__pinned_func
static inline int fuel_gauge_get_property(const struct device * dev, struct fuel_gauge_get_property * props, size_t props_len)
{
#ifdef CONFIG_USERSPACE
	if (z_syscall_trap()) {
		union { uintptr_t x; const struct device * val; } parm0 = { .val = dev };
		union { uintptr_t x; struct fuel_gauge_get_property * val; } parm1 = { .val = props };
		union { uintptr_t x; size_t val; } parm2 = { .val = props_len };
		return (int) arch_syscall_invoke3(parm0.x, parm1.x, parm2.x, K_SYSCALL_FUEL_GAUGE_GET_PROPERTY);
	}
#endif
	compiler_barrier();
	return z_impl_fuel_gauge_get_property(dev, props, props_len);
}
```